chia/cmds/db_upgrade_func.py

```diff
@@ -40,15 +46,64 @@ def db_upgrade_func(
         out_db_path = path_from_root(root_path, db_path_replaced)
         mkdir(out_db_path.parent)
 
-    convert_v1_to_v2(in_db_path, out_db_path)
-
-    if update_config:
-        print("updating config.yaml")
-        config = load_config(root_path, "config.yaml")
-        new_db_path = db_pattern.replace("_v1_", "_v2_")
-        config["full_node"]["database_path"] = new_db_path
-        print(f"database_path: {new_db_path}")
-        save_config(root_path, "config.yaml", config)
+    total, used, free = shutil.disk_usage(out_db_path.parent)
+    in_db_size = in_db_path.stat().st_size
+    if free < in_db_size:
+        no_free: bool = free < in_db_size * 0.6
+        strength: str
+        if no_free:
+            strength = "probably not enough"
+        else:
+            strength = "very little"
+        print(f"there is {strength} free space on the volume where the output database will be written:")
+        print(f"   {out_db_path}")
+        print(
+            f"free space: {free / 1024 / 1024 / 1024:0.2f} GiB expected about "
+            f"{in_db_size / 1024 / 1024 / 1024:0.2f} GiB"
+        )
+        if no_free and not force:
+            print("to override this check and convert anyway, pass --force")
+            return
+
+    try:
+        convert_v1_to_v2(in_db_path, out_db_path)
+
+        if update_config:
+            print("updating config.yaml")
+            config = load_config(root_path, "config.yaml")
+            new_db_path = db_pattern.replace("_v1_", "_v2_")
+            config["full_node"]["database_path"] = new_db_path
+            print(f"database_path: {new_db_path}")
+            save_config(root_path, "config.yaml", config)
+
+    except RuntimeError as e:
+        print(f"conversion failed with error: {e}.")
+    except Exception as e:
+
+        print(
+            textwrap.dedent(
+                f"""
```

Comment:
Unless you want the leading blank line.  Indentation is a bit inconsistent around here.

```suggestion
                f"""\
```

Reply:
aha! I was wondering if there was a way out of this. `dedent()` required me to have the indentation on the first line as well.